### DISCUSSION TITLE
How can I use it on multiple machines (with same "personality")? - OR: where are "my" list of shares stored?

### DISCUSSION THREAD

**Discussion by tschloss** [robherley/snips.sh#171] [Q&A]
Hi, just discoverd this great tool!

What I did not understand ist how the tui knows all of my shares. Do I identify myself against the service or is the list of shares stored locally?

Background: I want to use it on multiple machines with one "repository".

Thank you
Thomas

**Comment by tschloss** [comment]
Oh I forgot that ssh sends my identity automatically. So with ssh -i identity snips.sh I can change my identity.

**Comment by robherley** [comment]
👋 Thanks for the kind words @tschloss 

And you are correct, users are identified by using the fingerprint of the SSH public key.

Internally they are stored as following:
https://github.com/robherley/snips.sh/blob/8e52e02c66e74479d5643d22a2c6490cd73a5b7f/internal/db/schema.hcl#L23-L53

If a new user connects (new public key not stored in database) they will automatically have a new user id assigned.

A similar discussion (and feature request) to map multiple keys to a single user is here:
- https://github.com/robherley/snips.sh/discussions/164

Having a single identity for snips totally works 👍
